lib/facter/self_service.rb

```diff
@@ -122,6 +122,20 @@
     # check for use_cached_catalog logic flip as false is the desired state
     { S0030: !Puppet.settings['use_cached_catalog'] }
   end
+
+  chunk(:S0034) do
+    # PE has not been upgraded / updated in 1 year
+    # It was decided not to include infra components as this was deemed unecessary as they should align with the primary.
+
+    # gets the file for the most recent upgrade output
+    last_upgrade_file = Dir.glob('/var/log/puppetlabs/installer/*.install.log').max_by { |f| File.mtime(f) }
+
+    # get the timestamp for the most recent upgrade
+    last_upgrade_time = File.mtime(last_upgrade_file)
```

Comment:
This doesn't handle a case where there are no install logs, which is causing the tests to fail. You may want to check that `last_upgrade_file` is what you expect it to be after line 131.

Reply:
Something like:

```
    next unless PuppetSelfService.primary?

    # gets the file for the most recent upgrade output
    last_upgrade_file = Dir.glob('/var/log/puppetlabs/installer/*.install.log').max_by { |f| File.mtime(f) }

    if last_upgrade_file.nil? 
      { S0034: false }
   else
    # get the timestamp for the most recent upgrade
    last_upgrade_time = File.mtime(last_upgrade_file)
    # last upgrade was sooner than 1 year ago
    { S0034: last_upgrade_time >= (Time.now - 31_536_000).utc }
  end
```